builtin/logical/pki/path_fetch.go

```diff
@@ -261,6 +262,106 @@ func (b *backend) pathFetchCertList(ctx context.Context, req *logical.Request, d
 	return logical.ListResponse(entries), nil
 }
 
+func pathFetchListCertsDetailed(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: "certs/detailed/?$",
+
+		DisplayAttrs: &framework.DisplayAttributes{
+			OperationPrefix: operationPrefixPKI,
+			OperationSuffix: "certs",
+		},
+
+		Fields: map[string]*framework.FieldSchema{
+			"after": {
+				Type:        framework.TypeString,
+				Description: `Optional entry to list begin listing after, not required to exist.`,
+			},
+			"limit": {
+				Type:        framework.TypeInt,
+				Description: `Optional number of entries to return; defaults to all entries.`,
+			},
+		},
+
+		Operations: map[logical.Operation]framework.OperationHandler{
+			logical.ListOperation: &framework.PathOperation{
+				Callback: b.pathFetchCertListDetailed,
+				Responses: map[int][]framework.Response{
+					http.StatusOK: {{
+						Description: "OK",
+						Fields: map[string]*framework.FieldSchema{
+							"keys": {
+								Type:        framework.TypeStringSlice,
+								Description: `A list of keys`,
+								Required:    true,
+							},
+							"key_info": {
+								Type:        framework.TypeMap,
+								Description: `Key info with certificate details`,
+								Required:    false,
+							},
+						},
+					}},
+				},
+			},
+		},
+
+		HelpSynopsis:    pathFetchHelpSyn,
+		HelpDescription: pathFetchHelpDesc,
+	}
+}
+
+func (b *backend) pathFetchCertListDetailed(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	var responseKeys []string
+	responseInfo := make(map[string]interface{})
+
+	after := data.Get("after").(string)
+	limit := data.Get("limit").(int)
+	if limit <= 0 {
+		limit = -1
+	}
+
+	entries, err := req.Storage.ListPage(ctx, "certs/", after, limit)
+	if err != nil {
+		return nil, err
+	}
+	for i := range entries {
+		// Fetch the full certificate entry by key
+		entry, err := req.Storage.Get(ctx, "certs/"+entries[i])
+		if err != nil {
+			return nil, err
+		}
+		if entry == nil {
+			return logical.ErrorResponse(fmt.Sprintf("failed to retrieve entry for %s", entries[i])), nil
+		}
+
+		entries[i] = denormalizeSerial(entries[i])
+		responseKeys = append(responseKeys, string(entries[i]))
+
+		// Parse the certificate details
+		certData, err := x509.ParseCertificate(entry.Value)
+		if err != nil {
+			return logical.ErrorResponse(fmt.Sprintf("failed to parse certificate for %s: %s", entries[i], err)), nil
+		}
+
+		// limit DNS names to 5
+		dnsNames := certData.DNSNames
+		if len(dnsNames) > 5 {
+			dnsNames = dnsNames[:5]
+		}
+
+		responseInfo[string(entries[i])] = map[string]interface{}{
+			"common_name": certData.Subject.CommonName,
```

Comment:
Trying to decide if I like just `common_name` (which is, I guess, most common), or if the full `subject` would be preferable?

What's your take?

Reply:
Hmm the full [subject](https://cs.opensource.google/go/go/+/refs/tags/go1.23.3:src/crypto/x509/pkix/pkix.go;l=123-139) does have a lot of details which includes String lists of Country, Organization, OrganizationalUnit, Locality, Province, StreetAddress, PostalCode &c.  In the tests I wrote, the only subject details returned was the CN, I'm not quite sure how often these are filled out and how long the lists typically are.

Maybe restricting each list to <= 5 names if they exist, OR to return just the first 5 subject details in cases where more than 5 are present, OR both. 

Let me know your thoughts!

Reply:
Yeah, that is true that Subject can grow long. A lot of organizations like setting it, though many components will be the same (except maybe O/OU)... Let's stick with `common_name` for now and someone can always file a RFE later and we can add additional fields then. :-)

Reply:
Alrighty, it is a simple add so if anyone does request it, we can always add it :)